sources/SilkTouch/SilkTouch/Mods/TransformCOM.cs

```diff
@@ -0,0 +1,749 @@
+                    }
+                }
+
+                return base.VisitBinaryExpression(node);
+            }
+
+            public override SyntaxNode? VisitVariableDeclaration(VariableDeclarationSyntax node)
+            {
+                if (!node.Variables.Any(var => var.Initializer is not null && var.Initializer.Value is LiteralExpressionSyntax lit && lit.IsKind(SyntaxKind.NullLiteralExpression)))
+                    return base.VisitVariableDeclaration(node);
+
+                // Get the corresponding parameter type
+                var variableType = node.Type;
+                var name = GetTypeName(variableType?.ToString() ?? string.Empty);
+
+                // Check if the parameter type is ComType
+                if (!ComTypes.Any(type => $"{type.Item1}*" == name))
+                {
+                    return base.VisitVariableDeclaration(node);
+                }
+
+                var newInstantiation = node;
+                var variables = node.Variables;
+                for (int i = 0; i < variables.Count; i++)
+                {
+                    var variableExpression = variables[i].Initializer?.Value as LiteralExpressionSyntax;
+                    // Check if the variable is instantiated to a null literal
+                    if (variableExpression is not null && variableExpression.IsKind(SyntaxKind.NullLiteralExpression))
+                    {
+                        var newDefaultArg = LiteralExpression(SyntaxKind.DefaultLiteralExpression);
+                        newInstantiation = newInstantiation.ReplaceNode(variableExpression, newDefaultArg);
+                    }
+                }
+                return newInstantiation;
+            }
+
+            private string GetTypeName(string fullTypeName)
+            {
+                int dot = fullTypeName.LastIndexOf('.');
+
+                if (dot != -1)
+                {
+                    return fullTypeName.Substring(dot + 1);
+                }
+
+                return fullTypeName;
+            }
+        }
+
+        class Rewriter(List<(string, bool, KeyedStringTree?)> ComTypes)
+            : CSharpSyntaxRewriter
+        {
+            public override SyntaxNode? VisitPointerType(PointerTypeSyntax node)
+            {
+                for (int i = 0; i < ComTypes.Count; i++)
+                {
+                    (string, bool, KeyedStringTree?) val = ComTypes[i];
+
+                    if (val.Item1 == node.ElementType.ToString())
+                    {
+                        return IdentifierName(val.Item1);
+                    }
+                }
+
+                return base.VisitPointerType(node);
+            }
+
+            public override SyntaxNode? VisitIdentifierName(IdentifierNameSyntax node)
+            {
+                if (node.Identifier.ToString() == "lpVtbl")
+                {
+                    return ParenthesizedExpression(PrefixUnaryExpression(SyntaxKind.PointerIndirectionExpression, node));
```

Comment:
Couldn't this end up with `someInterface.lpVtbl` -> `someInterface.(*lpVtbl)`? Do we care?

Reply:
Theoretically, yes. Haven't seen it happen in practice since lpVtbl is a ClangSharp added variable so nothing should access it